cpp/ql/test/query-tests/Security/CWE/CWE-079/semmle/CgiXss/search.c

```diff
@@ -37,14 +37,26 @@ void good_server(char* query) {
   puts(do_search(query));
 }
 
+int snprintf(char *, int, const char *, ...);
+
+void good_server2(char* query) {
+  puts("<p>Query results for ");
+  // GOOD: Only an integer is added to the page.
+  int i = 0;
+  snprintf(query, 16, "value=%i", &i);
+  printf("\n<p>%i</p>\n", i);
```

Comment:
Would this have produced a false positive result without the fix in this PR?  Is `snprintf` supposed to be `scanf`???

Reply:
Oops, you're right. Good catch! I think I mixed up my test cases. Indeed, this new test does not produce a false positive on `main` currently. It does if I replace `snprintf` with `sscanf` (which is what I wanted to write). I've fixed this in f5d5460dde1768b82c7ac21bd022d99f8d3e427e.